### PR TITLE
Mailer

### PR DESCRIPTION
**Pull request by joaquinvalentin**

### CODE REVIEW

**Review by MarcosFierro — APPROVED** [review]

File: app/admin/users.rb [hunk 1]
```diff
@@ -4,13 +4,18 @@
   permit_params :email, :password, :admin
 
   before_action :remove_password_params_if_blank, only: [:update]
+  after_action :send_email, only: [:create]
```
Comment (guillezorron) [inline app/admin/users.rb]: Why not using a `after_create` callback in the model? If we use it we do not need to do this here and in the controller.

File: app/admin/users.rb [hunk 2]
```diff
@@ -4,13 +4,18 @@
...
   controller do
     def remove_password_params_if_blank
       return unless params[:user][:password].blank? && params[:user][:password_confirmation].blank?
 
       params[:user].delete(:password)
       params[:user].delete(:password_confirmation)
     end
+
+    def send_email
+      UserMailer.with(user: params[:user], admin: current_user.email).welcome_email.deliver_now if user.valid?
```
Comment (guillezorron) [inline app/admin/users.rb]: If you use a `after_create` you don't need to check if the user is valid

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -15,7 +15,10 @@ def create
     authorize user
     return render_error(4106) if user_by_email
 
-    return render json: UserSerializer.render(user), status: :created if user.save
+    if user.save
```
Comment (guillezorron) [inline app/controllers/api/v1/users_controller.rb]: Please remove it and do it in the model

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -67,4 +70,8 @@ def check_permissions
   def check_login
     render_error(4107) unless current_user
   end
+
+  def send_email(user)
```
Comment (guillezorron) [inline app/controllers/api/v1/users_controller.rb]: Same comment

File: config/initializers/smtp.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+ActionMailer::Base.smtp_settings = {
+  domain: 'jv-training.herokuapp.com',
+  address: 'smtp.sendgrid.net',
+  port: 587,
+  authentication: :plain,
+  user_name: 'apikey',
+  password: ENV['SENDGRID_API_KEY']
```
Comment (guillezorron) [inline config/initializers/smtp.rb]: This env variable should be on the application.rb with the fetch method to check it

File: config/initializers/smtp.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+ActionMailer::Base.smtp_settings = {
+  domain: 'jv-training.herokuapp.com',
+  address: 'smtp.sendgrid.net',
+  port: 587,
+  authentication: :plain,
+  user_name: 'apikey',
```
Comment (guillezorron) [inline config/initializers/smtp.rb]: All this values should be environment variables

File: app/admin/users.rb (hunk 1 above)
Comment (joaquinvalentin) [reply app/admin/users.rb]: I need to send the email from the creator email, and using after_create I probably need to change the model to save that email to use it, right? 
Do you think it is important to do this now? Anyway, I understand your approach and think it is better

**Pull request merged**